api/src/main/java/jakarta/data/metamodel/NavigableExpression.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2025 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.metamodel;
+
+public interface NavigableExpression<T,U> {
+
+    default <V> NavigableExpression<T, V> navigate(NavigableAttribute<V, V> attribute) {
+        throw new UnsupportedOperationException("not yet implemented");
+    }
+
+    default TextExpression<T> navigate(TextAttribute<U> attribute) {
```

Comment:
It is unfortunate that this will require the user to input information that should already be known,

```java
_Book.author.navigate(_Author.name).equalTo(authorName);
```

But I haven't been able to think of any way around it.  The closest I could come up with is to have the metamodel class supply the attribute (`_Author`) when constructing the NavigableAttribute, and have,
```
_Book.author.navigate().equalTo(authorName);
```
although this doesn't work due to warnings about invoking instance methods on a static class and we would also lose the information about which entity the association came from.

Reply:
the only way I can see to do better would be to generate additional types, something approximately like this:

```java
@StaticMetamodel(Book.class)
@Generated("org.hibernate.processor.HibernateProcessor")
public interface _Book {
    TextAttribute<Book,String> isbn = ... ;
    TextAttribute<Book,String> title = ... ;

    _Author.Navigable<Book>  authors = .... ; 

    interface Navigable<T> extends NavigableAttribute<T,Book> {
        Expression<Book,String> isbn = ... ;
        Expression<Book,String> title = ... ;
        ...
    }
}
```

So you could write `_Book.authors.name.equalTo(authorName)` or `_Author.books.title.startsWith("Jakarta")`.

I suppose that could be made to work but it does mean generating that extra type.

Reply:
(Perhaps `NavigableAttribute` is not needed in this approach. Need to prototype something.)